File: HTML5.ScratchPad.AngularGrid.Web/app/domain/Address.ts
///<reference path="IAddress.ts" />

module app.domain {

    export class Address extends app.domain.EntityBase implements app.domain.IAddress {
        constructor(
            public AddressLine1: string,
            public AddressLine2: string,
            public AddressLine3: string,
            public AddressLine4: string,
            public Email: string,
            public Postcode: string,
            public AddressId?: number
            ) {

            super();
            
            //Relative properties
            this.AddressId = AddressId;
            this.AddressLine1 = AddressLine1;
            this.AddressLine2 = AddressLine2;
            this.AddressLine3 = AddressLine3;
            this.AddressLine4 = AddressLine4;
            this.Email = Email;
            this.Postcode = Postcode;
        }
    }
}

File: HTML5.ScratchPad.DDD.Domain/Entities/Address.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HTML5.ScratchPad.DDD.Domain.Entities
{
    public class Address
    {
        [Key]
        //[Key, ForeignKey("AddressId")]
        public int AddressId { get; set; }

        [Required]
        public string AddressLine1 { get; set; }
        public string AddressLine2 { get; set; }
        public string AddressLine3 { get; set; }
        public string AddressLine4 { get; set; }

        [Required]
        public string Postcode { get; set; }

        public string Email { get; set; }

        //public virtual int CustomerId { get; set; }
        //[Key, ForeignKey("CustomerId")]

        public virtual Customer Customer { get; set; }
    }
}
